docs/madr/decisions/035-mesh-metric-policy.md

```diff
@@ -0,0 +1,126 @@
+We want to create a new policy responsible for Metrics collection. This policy will be part of Kuma observability policies 
+stack, alongside `MeshAccessLog` and `MeshTrace`. The new policy should expose the same functionality as now is exposed through `Mesh`. 
+We want to provide support for Prometheus and native support for OTEL backend.
+
+## Decision Outcome
+We will implement new `MeshMetric` policy as described further in this MADR.
+
+## Solution
+
+### API design
+
+#### Current API
+The API is going to be similar to what we have under `mesh.spec.metrics`. The current API looks like this:
+
+```yaml
+apiVersion: kuma.io/v1alpha1
+kind: Mesh
+metadata:
+  name: default
+spec:
+  metrics:
+  enabledBackend: prometheus-1
+  backends:
+    - name: prometheus-1
+      type: prometheus
+      conf:
+        tls:
+          mode: disabled | providedTLS | activeMTLSBackend
+        port: 5670
+        path: /metrics
+        tags:
+          kuma.io/service: dataplane-metrics
+        envoy:
+          filterRegex: http2_act.*
+          usedOnly: true
+        aggregate:
+          - name: my-service
+            path: "/metrics/prometheus"
+            port: 8888
+          - name: other-sidecar
+            port: 8000
+```
+
+#### Changes to the API
+1. I think the aggregation that [uses the aggregate name and pod annotations](https://kuma.io/docs/2.5.x/policies/traffic-metrics/#expose-metrics-from-applications) no longer makes sense, as we can use targetRef 
+policies to point to specific DPs. I imagine there would be a default wide policy (possibly `Mesh` scoped) and per app / framework / language overrides.
+2. What I also don’t like about this is that “filters” are under envoy - I’d probably rename it to filter and introduce 
+the concept of “profiles” like we discussed. Maybe something like this:
+
+```yaml
+filter:
+  usedOnly: true # true or false
+  profile: minimal # one of minimal, default, full
+  regex: http2_act.* # only profile or regex can be defined
+```
+3. Should we also think about grouping the settings somehow, tls/port/path/tags/envoy are all “mesh/sidecar” metrics and 
+“aggregate” are application metrics, or maybe we should just rename `aggregate` to `otherContainers`?
+
+#### What other meshes do
+- Istio has [Telemetry](https://istio.io/latest/docs/reference/config/telemetry/#Metrics) policy that allows defining providers (our backends), overrides (that allows disabling / changing tags), reporting interval
+- LinkerD has only tutorials on how to [export metrics](https://linkerd.io/2.14/tasks/exporting-metrics/#proxy), and creating a service profile can enable per route metrics
+
+#### New API suggestion:
+
+```yaml
+apiVersion: kuma.io/v1alpha1
+kind: MeshMetrics
+metadata:
+  name: default
+  namespace: kuma-system
+spec:
+  targetRef:
+    kind: Mesh | MeshSubset | MeshService | MeshServiceSubset
+  default:
+    filter:
+      profile: minimal
+      regex: http2_act.* # only profile or regex can be defined
```

Comment:
can this be a list?

Reply:
this cannot be a list, it is rewritten to: https://www.envoyproxy.io/docs/envoy/latest/operations/admin#get--stats?filter=regex

Reply:
ok